pkg/operator/starter.go

```diff
@@ -175,13 +176,20 @@ func RunOperator(ctx context.Context, controllerContext *controllercmd.Controlle
 		WithEventHandler(operatorclient.TargetNamespace, "LateConnections", terminationobserver.ProcessLateConnectionEvents).
 		ToController(kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace), kubeClient.CoreV1(), controllerContext.EventRecorder)
 
-	infrastructure, err := configClient.ConfigV1().Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
 	var notOnSingleReplicaTopology resourceapply.ConditionalFunction = func() bool {
-		return infrastructure.Status.ControlPlaneTopology != configv1.SingleReplicaTopologyMode
```

Comment:
let's not change this function in this PR, we cannot hide errors, we need a better strategy.

Reply:
instead of hiding the error we could change the controller that accepts this method to report it.
the controller could also wait until the preconditions will be fulfilled before creating/deleting the resources.

Reply:
yes, revert the change and add the TODO.